velox/exec/SortBuffer.cpp

```diff
@@ -287,9 +287,11 @@ void SortBuffer::spillOutput() {
 void SortBuffer::prepareOutput(uint32_t maxOutputRows) {
   VELOX_CHECK_GT(maxOutputRows, 0);
   VELOX_CHECK_GT(numInputRows_, numOutputRows_);
-
+  const uint64_t batchSizeMaybe = numInputRows_ - numOutputRows_;
   const vector_size_t batchSize =
-      std::min<vector_size_t>(numInputRows_ - numOutputRows_, maxOutputRows);
+      batchSizeMaybe > std::numeric_limits<vector_size_t>::max()
+      ? maxOutputRows
+      : std::min<vector_size_t>(batchSizeMaybe, maxOutputRows);
```

Comment:
we can use std::min<uint64_t>(batchSizeMaybe, maxOutputRows) since have the check above.